deepspeed/runtime/zero/stage_1_and_2.py

```diff
@@ -1630,6 +1719,9 @@ def get_flat_partition(self,
             # we need to offset to get to the right element
             if i == 0 and first_offset > 0:
                 tensor_offset = first_offset
+                if tensor_offset > num_elements:
```

Comment:
I have two questions here:
1. How can `tensor_offset > num_elements` be True? Does this mean the first tensor is skipped entirely from the flattened result?
2. Line 1723 seems to be a noop because of 1717?

Reply:
This is because the first_offset maybe not corresponding to the first tensor of the group. So ` tensor_offset > num_elements`  can be True. I found that this would set every `tensor.grad ` a wrong position, although the total tensor size is right. I will fix it asap.

Reply:
Two more questions:
1. So, it is possible for the first tensor to be missing from the flattened version of the partition. Is this correct? Previously, every tensor (sometimes a subset of the first tensor) was included in the flattened version. 

2. Also, is it only the first tensor that could be missing from the flattened version, or could other tensors be missing?

Reply:
No. Every tensor should be included in the flattened version I think.

Reply:
Sorry, I still don't get it :). Line 1724 will exclude the first tensor from the flattened version, no?

Reply:
The design on Line 1724 will be deprecated. I am working on a new design :).

Reply:
Got it. Thanks so much!